### ISSUE TITLE
Some tree items are not shown in the correct path

### ISSUE THREAD

**Issue by alp82** [react-cosmos/react-cosmos#495]
### What's up?

I got a folder with multiple subfolders with one component each. All of those components have one or more fixtures. They show up in react-cosmos as one would expect, except for one which is shown at the root instead of a subfolder.

### Mkay, tell me more...

Here is the folder structure (ui/grid is the important part):
![image](https://user-images.githubusercontent.com/966342/32415197-fa2f1eae-c235-11e7-826f-fdd2cda1c2a4.png)

This is the resulting view in react-cosmos (again look at Grid):
![image](https://user-images.githubusercontent.com/966342/32415217-4e18bd72-c236-11e7-8664-ec000e2242d5.png)

The `Grid` component with its `nested` fixture is placed at the top for no apparent reason.

**Comment by skidding** [comment]
It surprises me that this still happens with 3.x. I remember this only happened before the major fixture factor. To make it easier for me to debug this please run this (new) command and let me know what the output is:

```bash
node_modules/.bin/print-fixture-files
# If on a Mac
node_modules/.bin/print-fixture-files | pbcopy
```

Optionally, for your privacy:
- Replace the root path with a foo path
- Send the output to me via IM

Note: `print-fixture-files` is only available since beta.7 ([see this comment](https://github.com/react-cosmos/react-cosmos/issues/494#issuecomment-341986021))

**Comment by alp82** [comment]
Here you go. I grepped for the grid folder only:
```
[
  {
    "filePath": "/my-project/ui/grid/Flex/__cosmox__/bigger.js",
    "components": [
      {
        "name": "Flex",
        "filePath": "/my-project/ui/grid/Flex/Flex.js"
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/Flex/__cosmox__/huge-spaces.js",
    "components": [
      {
        "name": "Flex",
        "filePath": "/my-project/ui/grid/Flex/Flex.js"
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/Flex/__cosmox__/smaller.js",
    "components": [
      {
        "name": "Flex",
        "filePath": "/my-project/ui/grid/Flex/Flex.js"
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/FlexItem/__cosmox__/default.js",
    "components": [
      {
        "name": "FlexItem",
        "filePath": "/my-project/ui/grid/FlexItem/FlexItem.js"
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/Grid/__cosmox__/nested.js",
    "components": [
      {
        "name": "Grid",
        "filePath": null
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/GridItem/__cosmox__/simple.js",
    "components": [
      {
        "name": "GridItem",
        "filePath": "/my-project/ui/grid/GridItem/GridItem.js"
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/Layout/__cosmox__/combined.js",
    "components": [
      {
        "name": "Layout",
        "filePath": null
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/Layout/__cosmox__/main.js",
    "components": [
      {
        "name": "Layout",
        "filePath": "/my-project/ui/grid/Layout/Layout.js"
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/Layout/__cosmox__/sidebar.js",
    "components": [
      {
        "name": "Layout",
        "filePath": "/my-project/ui/grid/Layout/Layout.js"
      }
    ]
  },
  {
    "filePath": "/my-project/ui/grid/LayoutItem/__cosmox__/simple.js",
    "components": [
      {
        "name": "LayoutItem",
        "filePath": "/my-project/ui/grid/LayoutItem/LayoutItem.js"
      }
    ]
  },
]
```

`"filePath": null` looks suspicious.

Here is the `nested.js` fixture:
```
import React from 'react';
import GridItem from 'ui/grid/GridItem';
import Grid from 'ui/grid/Grid';

const childCss1 = {
  backgroundColor: 'red',
  textAlign: 'center',
  height: '5em',
};

const childCss2 = {
  backgroundColor: 'green',
  textAlign: 'center',
  height: '5em',
};

export default {
  component: Grid,
  children: [
    <GridItem key="1" css={childCss1} colFrom="1" colTo="2" rowFrom="1" rowTo="1">
      cell 1
    </GridItem>,
    <GridItem key="2" css={childCss1} colFrom="2" colTo="3" rowFrom="1" rowTo="1">
      cell 2
    </GridItem>,
    <GridItem key="3" css={childCss1} colFrom="3" colTo="4" rowFrom="1" rowTo="1">
      cell 3
    </GridItem>,
    <GridItem key="4" css={childCss1} colFrom="4" colTo="5" rowFrom="1" rowTo="1">
      cell 4
    </GridItem>,
    <GridItem key="5" css={childCss1} colFrom="5" colTo="6" rowFrom="1" rowTo="1">
      cell 5
    </GridItem>,
    <GridItem key="6" css={childCss1} colFrom="6" colTo="11" rowFrom="1" rowTo="1">
      <Grid>
        <GridItem key="n1" css={childCss2} colFrom="1" colTo="2" rowFrom="1" rowTo="1">
          nested cell 1
        </GridItem>
        <GridItem key="n2" css={childCss2} colFrom="2" colTo="3" rowFrom="1" rowTo="1">
          nested cell 2
        </GridItem>
        <GridItem key="n3" css={childCss2} colFrom="3" colTo="7" rowFrom="1" rowTo="1">
          nested cell 3
        </GridItem>
        <GridItem key="n4" css={childCss2} colFrom="7" colTo="9" rowFrom="1" rowTo="1">
          nested cell 4
        </GridItem>
        <GridItem key="n5" css={childCss2} colFrom="9" colTo="11" rowFrom="1" rowTo="1">
          nested cell 5
        </GridItem>
      </Grid>
    </GridItem>,
  ],
};
```

And `combined.js`:
```
import React from 'react';
import Layout from 'ui/grid/Layout';
import LayoutItem from 'ui/grid/LayoutItem';

export default {
  component: Layout,
  props: { template: 'main' },
  children: [
    <LayoutItem key="1" name="header" css={{ backgroundColor: 'orange' }}>
      header
    </LayoutItem>,
    <LayoutItem key="2" name="content" css={{ backgroundColor: 'lightblue' }}>
      <Layout template="sidebar">
        <LayoutItem key="s1" name="primary" css={{ backgroundColor: 'red' }}>
          primary
        </LayoutItem>
        <LayoutItem key="s2" name="secondary" css={{ backgroundColor: 'yellow' }}>
          secondary
        </LayoutItem>
      </Layout>
    </LayoutItem>,
    <LayoutItem key="3" name="footer" css={{ backgroundColor: 'lightgreen' }}>
      footer
    </LayoutItem>,
  ],
};
```

I think it's quite obvious that the problem is that i use the component which is about to be tested as a nested child. Any ideas?

**Comment by skidding** [comment]
I think it's from the import path. Two questions:
- Are the nested and combined fixtures the only ones where you use import path aliases? (eg. `ui/grid/Grid`)
- Are you using babel-plugin-module-resolver?

**Comment by alp82** [comment]
* I don't use any aliases. all my components use that form of importing. Maybe this is related:
```
// webpack.config.js
// ...
  resolve: {
    modules: [
      path.resolve(__dirname),
      'node_modules',
    ],
    extensions: ['.js'],
  },
```
* no. here is the babel config:
```
{
  "presets": [
    "es2015",
    "react",
    "stage-1",
    "env"

  ],
  "plugins": [
    "react-hot-loader/babel",
    "transform-class-properties",
    "transform-react-jsx-source",
    "glamorous-displayname",
    "wildcard"
  ],
  "env": {
    "development": {
      "plugins": [
        ["transform-runtime", {
          "polyfill": true
        }]
      ]
    }
  }
}
```

**Comment by skidding** [comment]
> all my components use that form of importing

Interesting. I wonder how to other paths are detected...

Can you please show me an example of a fixture that *does* have its file path properly detected, so I can compared it with the ones that don't. Eg. `/my-project/ui/grid/GridItem/__cosmox__/simple.js`

**Comment by skidding** [comment]
@alp82 please try `react-cosmos@3.0.0-beta.10`

**Comment by alp82** [comment]
Sadly, i got an error page:
```
Error: relative path without a basedir
▼ 15 stack frames were expanded.
(anonymous function)
node_modules/commondir/index.js:15
./node_modules/commondir/index.js.module.exports
node_modules/commondir/index.js:13
getCommonDirFromPaths
node_modules/react-cosmos-voyager2/lib/client/get-components.js:197
getCollapsedComponentNamespace
node_modules/react-cosmos-voyager2/lib/client/get-components.js:205
_loop
node_modules/react-cosmos-voyager2/lib/client/get-components.js:138
getComponents
node_modules/react-cosmos-voyager2/lib/client/get-components.js:170
./node_modules/react-cosmos/lib/client/mount.js.exports.default
node_modules/react-cosmos/lib/client/mount.js:38
run
node_modules/react-cosmos/lib/client/loader-entry.js:11
./node_modules/react-cosmos/lib/client/loader-entry.js
node_modules/react-cosmos/lib/client/loader-entry.js:14
__webpack_require__
/my-project/webpack/bootstrap cd36c98bcc3e685f89c2:678
fn
/my-project/webpack/bootstrap cd36c98bcc3e685f89c2:88
0
http://localhost:8989/loader/main.js:148147:18
__webpack_require__
/my-project/webpack/bootstrap cd36c98bcc3e685f89c2:678
./app/configure-store.js.Object.defineProperty.value
/my-project/webpack/bootstrap cd36c98bcc3e685f89c2:724
(anonymous function)
http://localhost:8989/loader/main.js:728:10
▲ 15 stack frames were expanded.
This screen is visible only in development. It will not appear if the app crashes in production.
Open your browser’s developer console to further inspect this error.
```

Browser console:
```
Uncaught Error: relative path without a basedir
    at :8989/loader/main.js:17871
    at Array.reduce (<anonymous>)
    at ./node_modules/commondir/index.js.module.exports (:8989/loader/main.js:17869)
    at getCommonDirFromPaths (:8989/loader/main.js:83214)
    at getCollapsedComponentNamespace (:8989/loader/main.js:83222)
    at _loop (:8989/loader/main.js:83155)
    at getComponents (:8989/loader/main.js:83187)
    at ./node_modules/react-cosmos/lib/client/mount.js.exports.default (:8989/loader/main.js:83439)
    at run (:8989/loader/main.js:83379)
    at Object../node_modules/react-cosmos/lib/client/loader-entry.js (:8989/loader/main.js:83382)
    at __webpack_require__ (:8989/loader/main.js:679)
    at fn (:8989/loader/main.js:89)
    at Object.0 (:8989/loader/main.js:148147)
    at __webpack_require__ (:8989/loader/main.js:679)
    at ./app/configure-store.js.Object.defineProperty.value (:8989/loader/main.js:725)
    at :8989/loader/main.js:728
```

**Comment by skidding** [comment]
@alp82 Sorry about that! I think I've fixed it in #509. Please try `react-cosmos@3.0.0-beta.11`

**Comment by alp82** [comment]
beta12 fixed all problems related with this issue. Thank you!
